.github/workflows/staging-pipeline.yml

```
@@ -183,7 +183,109 @@
+            timeout 300 bash -c 'until [ "$(docker inspect --format="{{json .State.Health.Status}}" impact-graph-ql3)" == "\"healthy\"" ]; do echo "Waiting for impact-graph-ql3 to be healthy..."; sleep 5; done'
+            if [ $? -eq 124 ]; then
+                echo "Timeout waiting for impact-graph-ql3 to become healthy"
+                exit 1
+            fi
+            # Check if impact-graph-ql3 is healthy
+            if [ "$(docker inspect --format='{{json .State.Health.Status}}' impact-graph-ql3)" != "\"healthy\"" ]; then
+                echo "impact-graph-ql3 is not healthy, stopping deployment"
+                exit 1
+            fi
+            echo "Third deployment phase completed successfully"
+            echo "Full deployment completed successfully"
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)